types/react-refresh/runtime.d.ts

```diff
@@ -55,6 +55,11 @@ export function hasUnrecoverableErrors(): boolean;
  *    () => [useCustomHook], // Lazy to avoid triggering inline requires
  * );
  */
-export function createSignatureFunctionForTransform(): void;
+export function createSignatureFunctionForTransform(): <T>(
+    type: T,
+    key: string,
+    forceReset?: boolean,
+    getCustomHooks?: () => AnyFn[],
+) => T | undefined;
```

Comment:
maybe add void?

Reply:
I tried `void`, but `pnpm test react-refresh` complained not to use void here.

```
  63:10  error  void is only valid as a return type or generic type argument or the type of a `this` parameter  @typescript-eslint/no-invalid-void-type
```

Reply:
You can disable the rule is `void` is the correct type and needs to be used here.